src/SortMsg.chpl

```diff
@@ -29,89 +30,190 @@ module SortMsg
           }
         }
+
+        st.addEntry(rname, createSymEntry(sorted));
+        const repMsg = "created " + st.attrib(rname);
+        sortLogger.debug(getModuleName(),pn,getLineNumber(),repMsg);
+        return new MsgTuple(repMsg, MsgType.NORMAL);
+      }
+
+      select gEnt.dtype {
+        when DType.Int64 do return doSort(int);
+        when DType.UInt64 do return doSort(uint);
+        when DType.Float64 do return doSort(real);
+        otherwise {
+          var errorMsg = notImplementedError(pn,gEnt.dtype);
+          sortLogger.error(getModuleName(),pn,getLineNumber(), errorMsg);
+          return new MsgTuple(errorMsg, MsgType.ERROR);
+        }
       }
-      // Sort the input pda and create a new symbol entry for
-      // the sorted pda.
-      select (gEnt.dtype) {
-          when (DType.Int64) {
-              var e = toSymEntry(gEnt, int);
-              var sorted = doSort(e.a);
-              st.addEntry(sortedName, createSymEntry(sorted));
-          }// end when(DType.Int64)
-          when (DType.UInt64) {
-              var e = toSymEntry(gEnt, uint);
-              var sorted = doSort(e.a);
-              st.addEntry(sortedName, createSymEntry(sorted));
-          }// end when(DType.UInt64)
-          when (DType.Float64) {
-              var e = toSymEntry(gEnt, real);
-              var sorted = doSort(e.a);
-              st.addEntry(sortedName, createSymEntry(sorted));
-          }// end when(DType.Float64)
-          otherwise {
-              var errorMsg = notImplementedError(pn,gEnt.dtype);
-              sortLogger.error(getModuleName(),getRoutineName(),getLineNumber(),
-                                                     errorMsg);
-              return new MsgTuple(errorMsg, MsgType.ERROR);
-          }            
-      }// end select(gEnt.dtype)
-
-      repMsg = "created " + st.attrib(sortedName);
-      sortLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),repMsg);      
+    }
+
+    // https://data-apis.org/array-api/latest/API_specification/generated/array_api.searchsorted.html#array_api.searchsorted
+    @arkouda.registerND
+    proc searchSortedMsg(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab, param nd: int): MsgTuple throws {
+      param pn = Reflection.getRoutineName();
+      const x1 = msgArgs.getValueOf("x1"),
+            x2 = msgArgs.getValueOf("x2"),
+            side = msgArgs.getValueOf("side"),
+            rname = st.nextName();
+
+      var gEntX1: borrowed GenSymEntry = getGenericTypedArrayEntry(x1, st),
+          gEntX2: borrowed GenSymEntry = getGenericTypedArrayEntry(x2, st);
+
+      if side != "left" && side != "right" {
+        throw getErrorWithContext(
+          msg="Unrecognized side: %s".doFormat(side),
+          lineNumber=getLineNumber(),
+          pn,
+          moduleName=getModuleName(),
+          errorClass="NotImplementedError"
+        );
+      }
+
+      // TODO: add support for Float32
```

Comment:
Why doesn't this work with `Float32` today?

Reply:
It could, but would require a separate code path that I didn't add yet. I think that work could be saved for when we start adding more support for all the Array API types, but probably isn't needed yet.

Reply:
Sounds good to me, thanks.